webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { startCase } from 'lodash';
+import { sprintf, translate as __ } from 'foremanReact/common/I18n';
+import { Button } from 'patternfly-react';
+import ForemanModal from 'foremanReact/components/ForemanModal';
+import { useForemanModal } from 'foremanReact/components/ForemanModal/ForemanModalHooks';
+import { CANCEL_MODAL, RESUME_MODAL } from '../../TasksTableConstants';
+
+export const ConfirmModal = ({
+  actionText,
+  actionState,
+  tasksActions,
+  actionType,
+  selectedRowsLen,
+  id,
+}) => {
+  const { setModalClosed } = useForemanModal({
+    id,
+  });
+  if ([CANCEL_MODAL, RESUME_MODAL].includes(actionType)) {
+    selectedRowsLen = 1;
+  }
+  return (
+    <ForemanModal
+      title={sprintf(__('%s Selected Tasks'), startCase(actionText))}
+      id={id}
+    >
+      {sprintf(
+        __(
+          `This will %(action)s %(number)s task(s), putting them in the %(state)s state. Are you sure?`
+        ),
+        { action: actionText, number: selectedRowsLen, state: actionState }
+      )}
+      <ForemanModal.Footer>
+        <Button onClick={setModalClosed}>{__('No')}</Button>
+        <Button
+          bsStyle="primary"
+          onClick={() => {
+            tasksActions[actionType]();
+            setModalClosed();
+          }}
```

Comment:
can you move this logic too into the parent `TasksTablePage`?
this way it will be more generic `onClick={onClick}`

Reply:
I think it's better to add it here because the action should always close the modal and adding it into the parent will mean adding it 4 times

Reply:
I am talking more about the `tasksActions[actionType]`,
yes, the setModalClosed can stay here.

Reply:
The parent doesn't have the actionType so I'll leave it here